packages/typespec-client-generator-core/src/example.ts

```diff
@@ -0,0 +1,637 @@
+    const method = methodQueue.pop()!;
+    if (method.kind === "clientaccessor") {
+      methodQueue.push(...method.response.methods);
+    } else {
+      // since operation could have customization in client.tsp, we need to handle all the original operation (exclude the templated operation)
+      let operation = method.__raw;
+      while (operation && operation.templateMapper === undefined) {
+        const operationId = resolveOperationId(context.program, operation).toLowerCase();
+        if (examples.has(operationId)) {
+          diagnostics.pipe(handleMethodExamples(context, method, examples.get(operationId)!));
+          break;
+        }
+        operation = operation.sourceOperation;
+      }
+    }
+  }
+  return diagnostics.wrap(undefined);
+}
+
+function handleMethodExamples<TServiceOperation extends SdkServiceOperation>(
+  context: TCGCContext,
+  method: SdkServiceMethod<TServiceOperation>,
+  examples: Record<string, LoadedExample>
+): [void, readonly Diagnostic[]] {
+  const diagnostics = createDiagnosticCollector();
+
+  if (method.operation.kind === "http") {
+    diagnostics.pipe(handleHttpOperationExamples(method.operation, examples));
+    if (method.operation.examples) {
+      if (!context.__httpOperationExamples) {
+        context.__httpOperationExamples = new Map();
+      }
+      context.__httpOperationExamples!.set(method.operation.__raw, method.operation.examples);
+    }
+  }
+
+  return diagnostics.wrap(undefined);
+}
+
+function handleHttpOperationExamples(
+  operation: SdkHttpOperation,
+  examples: Record<string, LoadedExample>
+) {
+  const diagnostics = createDiagnosticCollector();
+  operation.examples = [];
+
+  for (const [title, example] of Object.entries(examples)) {
+    const operationExample = {
+      kind: "http",
+      name: title,
+      description: title,
+      filePath: example.file.path,
+      parameters: diagnostics.pipe(
+        handleHttpParameters(
+          operation.bodyParam
+            ? [...operation.parameters, operation.bodyParam]
+            : operation.parameters,
+          example.data,
+          example.relativePath
+        )
+      ),
+      responses: diagnostics.pipe(
+        handleHttpResponses(operation.responses, example.data, example.relativePath)
+      ),
+      rawExample: example.data,
+    } as SdkHttpOperationExample;
+
+    operation.examples.push(operationExample);
+  }
+
+  return diagnostics.wrap(undefined);
+}
+
+function handleHttpParameters(
+  parameters: SdkHttpParameter[],
+  example: any,
+  relativePath: string
+): [SdkHttpParameterExample[], readonly Diagnostic[]] {
+  const diagnostics = createDiagnosticCollector();
+  const parameterExamples = [] as SdkHttpParameterExample[];
+  if ("parameters" in example && typeof example.parameters === "object") {
+    for (const name of Object.keys(example.parameters)) {
+      const parameter = parameters.find(
+        (p) => (p.kind !== "body" && p.serializedName === name) || p.name === name
```

Comment:
nit, just a suggestion, no action needed.

I assume you do `p.name === name` for BODY.

But in TypeSpec, now it is much harder to know the name of parameter of BODY.

Java implement would do a fallback to "body" in example JSON, as this is most observed.